totalContributors.py

```diff
@@ -0,0 +1,95 @@
+import os
+import pathlib
+import subprocess
+import sys
+from datetime import datetime
+from typing import Dict, List
+
+
+def add_log(dict: Dict[str, int], input: List[str]) -> None:
+    for dataset in input:
+        name = dataset.split("\t")[1]
+        value = int(dataset.split("\t")[0])
+        if name in dict:
+            dict[name] += value
+        else:
+            dict[name] = value
+
+
+def retrieve_log(repo: str, lower_version: str, upper_version: str) -> List[str]:
+    return subprocess.check_output(
+        ["git", "shortlog", "-s", lower_version + ".." + upper_version],
+        universal_newlines=True,
+        cwd=find_path(repo),
+    ).splitlines()
+
+
+def find_version(time: str, repo: str) -> str:
+    versions_list = subprocess.check_output(
+        ["git", "tag", "-l"], universal_newlines=True, cwd=find_path(repo)
+    ).splitlines()
+    for version in versions_list:
+        version_time = subprocess.check_output(
+            ["git", "log", "-1", "--format=%ai", version],
+            universal_newlines=True,
+            cwd=find_path(repo),
+        ).split()[0]
+        if datetime.strptime(version_time, "%Y-%m-%d") >= datetime.strptime(time, "%Y-%m-%d"):
+            return version
+    return version
+
+
+def find_path(repo: str) -> str:
+    return os.path.dirname(pathlib.Path().resolve()) + "/" + repo
+
+
+isHelpMethod = False
+# extract git version and time
+if len(sys.argv) > 1:
+    if sys.argv[1] == "-h" or sys.argv[1] == "--help":  # help method
```

Comment:
You should use `argparse` for argument parsing. You can find many examples in this repo if you grep for argparse. Though one example would be https://github.com/zulip/zulip/blob/master/tools/run-mypy (mostly check the code in tools/ or scripts/, and so, notice that it doesn't end with .py, starts with shebang, etc). I think your script should be in put in `tools/`.

Reply:
This may take me a while to implement, I am currently pretty busy with some other things but I'll be sure to work on this once I get the chance. Sorry!

Reply:
No worries. Zulip 5.0 won't be released until at least October, so you have plenty of time.